### PR TITLE
Fix rolling restart on Linux

### PR DESCRIPTION
**Pull request by estellesoulard**
##### SUMMARY

Fixes the rolling restart on Linux in which the the consul service `restart`  was happening too soon after the `consul leave`, which resulted in the playbook proceeding without the new election finalizing and a temporarily leaderless cluster.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
- Rolling restart consul on Unix

##### ADDITIONAL INFORMATION
- Since we are using a new variable `consul_pid_file`, it was replaced in all occurrences in the templates as well.

### CODE REVIEW

**Comment by bbaassssiiee** [comment]
Thanks for your contribution! This repo is looking for maintainers...

If feasible, please rebase this PR on master to drop that merge commit. That way we have a linear commit history.

File: tasks/leave_restart_consul.yml [hunk 1]
```diff
@@ -1,9 +1,23 @@
 ---
 
-- name: Consul leave
+- name: Check if consul is running
   delegate_to: "{{ rolling_restart_host }}"
-  ansible.builtin.command: "{{ consul_binary }} leave {% if consul_acl_enable %} -token {{ consul_acl_master_token }} {% endif %} -http-addr {{ consul_addresses.http }}:{{ consul_ports.http }}"
-  changed_when: true
+  stat:
```
Comment (bbaassssiiee) [inline tasks/leave_restart_consul.yml]: Please use the FQCN `ansible.builtin.stat`.

File: tasks/leave_restart_consul.yml [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
+    path: "{{ consul_pid_file }}"
+  register: consul_pid
+
+- block:
```
Comment (bbaassssiiee) [inline tasks/leave_restart_consul.yml]: Ansible-lint says you can improve the task key order to: when, block. Also please name the block.

**Review by bbaassssiiee — CHANGES_REQUESTED** [review]
Please take a look at the output of the checks that ran and amend your PR.

File: tasks/leave_restart_consul.yml (hunk 1 above)
Comment (estellesoulard) [reply tasks/leave_restart_consul.yml]: Done :white_check_mark:

File: tasks/leave_restart_consul.yml (hunk 2 above)
Comment (estellesoulard) [reply tasks/leave_restart_consul.yml]: Done :white_check_mark: thanks :)

**Comment by bbaassssiiee** [comment]
There is still one linting issue:

```
yaml[indentation]: Wrong indentation: expected 4 but found 2
tasks/leave_restart_consul.yml:12
Error: Wrong indentation: expected 4 but found 2

WARNING  The following filters were mocked during the run: json_query
Read documentation for instructions on how to ignore specific rule violations.

                Rule Violation Summary                
 count tag               profile rule associated tags 
     1 yaml[indentation] basic   formatting, yaml     
```

**Comment by bbaassssiiee** [comment]
Would you have time to also remove the directory `./molecule/centos-stream-8`?
Centos 8 Stream End-of-life happened to happen 4 days ago (31 May 2024) [https://endoflife.date/centos-stream](https://endoflife.date/centos-stream)

**Review by bbaassssiiee — APPROVED** [review]
Great work!

**Pull request merged**